### ISSUE TITLE
Introspection seems to have generated a Prisma schema but it's not stored on the file system

### ISSUE THREAD

**Issue by nikolasburk** [prisma/prisma#700]
This issue is related to: https://github.com/prisma/prisma2/issues/699 It doesn't apply to the current `preview013` release because there the introspection is broken for another reason, so I downgraded to `preview-12` where I'm now seeing this issue.

I've tried this both with the [SQL Murder Mystery](https://github.com/NUKnightLab/sql-mysteries) as well as with another [sample SQLite DB](https://www.sqlitetutorial.net/wp-content/uploads/2018/03/chinook.zip) and the problem happens with both.

When running `prisma2 init`, I can successfully connect to the DB and the introspection seems to be succesful:

![image](https://user-images.githubusercontent.com/4058327/66481186-9cedae00-eaa0-11e9-95d9-497946f34d79.png)

However, after having selected the **Demo script** in the last step, the command fails:

![image](https://user-images.githubusercontent.com/4058327/66481268-ca3a5c00-eaa0-11e9-8ce6-51d4ebd5a947.png)

This is the error message:

```
$ prisma2 init murder2

Preparing your demo script (TypeScript) ...

✔  Downloading the demo script from GitHub ...
⠋  Extracting content to murder2 ...
3  Installing dependencies with: `npm install` ...
 ERROR  Error during command execution

Schema parsing error: Error validating model "CrimeSceneReport": Exactly one field must be marked as the id field with the `@id` di
rective..
  -->  schema.prisma:6
   | 
 5 | 
 6 | model CrimeSceneReport {
 7 |   city        String?
   | 
error: Error validating model "FacebookEventCheckin": Exactly one field must be marked as the id field with the `@id` directive..
  -->  schema.prisma:29
   | 
28 | 
29 | model FacebookEventCheckin {
30 |   date      Int?
   | 
error: Error validating model "GetFitNowCheckIn": Exactly one field must be marked as the id field with the `@id` directive..
  -->  schema.prisma:38
   | 
37 | 
38 | model GetFitNowCheckIn {
39 |   checkInDate  Int?             @map("check_in_date")
   | 
error: Error validating model "Interview": Exactly one field must be marked as the id field with the `@id` directive..
  -->  schema.prisma:65
   | 
64 | 
65 | model Interview {
66 |   person     Person? @map("person_id")
   | 
error: Error validating model "Solution": Exactly one field must be marked as the id field with the `@id` directive..
  -->  schema.prisma:86
   | 
85 | 
86 | model Solution {
87 |   user  Int?
   | 

```

So, it seems to me that there already is a Prisma schema file that was generated and includes a data model. However, the `schema.prisma` that's now part of the new project that was created by the `init` command now only contains the sample schema:

```groovy
datasource db {
  provider = "sqlite"
  url      = "file:./dev.db"
}

generator photon {
  provider = "photonjs"
}

model User {
  id    String  @default(cuid()) @id
  email String  @unique
  name  String?
  posts Post[]
}

model Post {
  id        String   @default(cuid()) @id
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  published Boolean  @default(true)
  title     String
  content   String?
  author    User?
}
```

Is this a bug and despite the fact that there are issues with the data model, would it be possible to store it in the `schema.prisma`?

**Comment by pantharshit00** [comment]
Yes,looks like a bug to me as well. Before even though we are not supporting columns with IDs right now, we were printing the datamodel to the schema instead printing the defaults.

**Comment by divyenduz** [comment]
Internal note: Dropping on internal introspection

**Comment by divyenduz** [comment]
@nikolasburk  We [rewrote introspection](https://github.com/prisma/prisma2/releases) recently. Can you please see if this issue still exists in the latest alpha or preview?

**Comment by pantharshit00** [comment]
I currently still error out: 
```
> prisma2 introspect --url="file:sql-murder-mystery.db"

Introspecting …

Introspection failed: Introspected schema can't be parsed.
Introspected Schema:

datasource db {
  provider = "sqlite"
  url      = "file:sql-murder-mystery.db"
}
model crime_scene_report {
  city        String?
  date        Int?
  description String?
  type        String?
}

model drivers_license {
  age          Int?
  car_make     String?
  car_model    String?
  eye_color    String?
  gender       String?
  hair_color   String?
  height       Int?
  id           Int?     @id
  plate_number String?
  persons      person[] @relation(references: [license_id])
}

model person {
  address_number          Int?
  address_street_name     String?
  id                      Int?                     @id
  license_id              drivers_license?
  name                    String?
  ssn                     Int?
  facebook_event_checkins facebook_event_checkin[]
  interviews              interview[]
  get_fit_now_members     get_fit_now_member[]
}

model facebook_event_checkin {
  date       Int?
  event_id   Int?
  event_name String?
  person_id  person?
}

model interview {
  person_id  person?
  transcript String?
}

model get_fit_now_member {
  id                    String?                @id
  membership_start_date Int?
  membership_status     String?
  name                  String?
  person_id             person?
  get_fit_now_check_ins get_fit_now_check_in[]
}

model get_fit_now_check_in {
  check_in_date  Int?
  check_in_time  Int?
  check_out_time Int?
  membership_id  get_fit_now_member?
}

model income {
  annual_income Int?
  ssn           Int? @id
}

model solution {
  user  Int?
  value String?
}

Schema parsing

error: Error parsing attribute "@id": Fields that are marked as id must be required.
  -->  schema.prisma:20
   |
19 |   height       Int?
20 |   id           Int?     @id
   |

error: Error parsing attribute "@id": Fields that are marked as id must be required.
  -->  schema.prisma:28
   |
27 |   address_street_name     String?
28 |   id                      Int?                     @id
   |

error: Error parsing attribute "@id": Fields that are marked as id must be required.
  -->  schema.prisma:50
   |
49 | model get_fit_now_member {
50 |   id                    String?                @id
   |

error: Error parsing attribute "@id": Fields that are marked as id must be required.
  -->  schema.prisma:67
   |
66 |   annual_income Int?
67 |   ssn           Int? @id
   |
```

**Comment by janpio** [comment]
Which schema is this @pantharshit00?

**Comment by pantharshit00** [comment]
It's https://github.com/NUKnightLab/sql-mysteries as mentioned in the original post.

**Comment by divyenduz** [comment]
@pantharshit00 It errors but does it write the file to the file system? Which is the concern of the original issue? If yes, then this can be closed and we can track this error in a new issue.

**Comment by divyenduz** [comment]
Related issue: https://github.com/prisma/prisma2/issues/1368

**Comment by divyenduz** [comment]
Note that https://github.com/prisma/prisma2/issues/1368 just tracks the messaging of this error, once that is merged and we have a better error message. I will link a new issue tracking the underlying error and link it here to close this one.

**Comment by pantharshit00** [comment]
No, the cli doesn't write the schema to the file system. It proceeds us to the error reporting dialog box.

<img width="1222" alt="image" src="https://user-images.githubusercontent.com/22195362/72810347-66b8f080-3c83-11ea-9a3c-8b72c353d5c5.png">


Tested with latest alpha: `prisma2@2.0.0-alpha.584, binary version: 1a0db4166ffce810428c731661582371d3ab88c4`

**Comment by janpio** [comment]
This code was fully replaced. If this still happens, please open a new issue and we are happy to take a look. Thanks.
